### DISCUSSION TITLE
pre-render astro files before Tailwind JIT

### DISCUSSION THREAD

**Discussion by Soviut** [withastro/roadmap#410] [Proposal]
## Problem

I have an astro page that loops over an array to render several sections. Each section has some Tailwind dynamic variants for custom `group` classes. I tried to name the dynamic groups by appending the index of the loop to the `group-<name>` .

```jsx
{items.map((item, index) => {
  <section class:list={`group/${index}`}>
}}
```

However, the Tailwind JIT complains since it is looking in the `.astro` files and encounters the variables in the class list and assumes they're invalid characters for classes.

## Proposal

Have a render pipeline that Tailwind can plug in to after a page has been rendered to HTML. Then the JIT can operate on the rendered HTML to gather all its classes. This pipeline would need to work in both the dev server and build pipeline.

**Comment by kotx** [comment]
This would give Astro an advantage over other frameworks because it allows for dynamic classes, which is not normally supported with Tailwind. +1

**Comment by florian-lefebvre** [comment]
This will not be done. You need to use Tailwind safelisting like any other framework: https://tailwindcss.com/docs/detecting-classes-in-source-files#safelisting-specific-utilities
